presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPruneSampleColumns.java

```diff
@@ -0,0 +1,77 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.sql.planner.Symbol;
+import io.prestosql.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.prestosql.sql.planner.iterative.rule.test.PlanBuilder;
+import io.prestosql.sql.planner.plan.Assignments;
+import io.prestosql.sql.planner.plan.ProjectNode;
+import io.prestosql.sql.planner.plan.SampleNode;
+import org.testng.annotations.Test;
+
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.expression;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.node;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.strictProject;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.values;
+import static io.prestosql.sql.planner.plan.SampleNode.Type.SYSTEM;
+
+public class TestPruneSampleColumns
+        extends BaseRuleTest
+{
+    @Test
+    public void testNotAllInputsReferenced()
+    {
+        tester().assertThat(new PruneSampleColumns())
+                .on(p -> p.project(
+                        Assignments.identity(p.symbol("b")),
+                        p.sample(
+                                0.5,
+                                SYSTEM,
+                                p.values(p.symbol("a"), p.symbol("b")))))
+                .matches(
+                        strictProject(
+                                ImmutableMap.of("b", expression("b")),
+                                node(SampleNode.class,
+                                        strictProject(
+                                                ImmutableMap.of("b", expression("b")),
+                                                values("a", "b")))));
+    }
+
+    @Test
+    public void testAllOutputsReferenced()
+    {
+        tester().assertThat(new PruneSampleColumns())
+                .on(p -> p.project(
+                        Assignments.identity(p.symbol("a"), p.symbol("b")),
+                        p.sample(
+                                0.5,
+                                SYSTEM,
+                                p.values(p.symbol("a"), p.symbol("b"))))).doesNotFire();
+    }
+
+    private ProjectNode buildProjectedSample(PlanBuilder planBuilder, Predicate<Symbol> projectionFilter)
```

Comment:
This method is not used anywhere

Reply:
Thanks for the feedback !! Applied the changes